File: backend/app/Models/Exemplaire.php
<?php

namespace App\Models;

use CodeIgniter\Model;

class Exemplaire extends MyBaseModel
{
    protected $table            = 'exemplaire';
    protected $primaryKey       = 'id';
    protected $useAutoIncrement = false;
    protected $returnType       = 'array';
    protected $useSoftDeletes   = false;
    protected $protectFields    = true;
    protected $allowedFields    = [ 'id', 'numero', 'dateAchat', 'photo', 'idEtat' ];

    protected bool $allowEmptyInserts = true;

    // Dates
    protected $useTimestamps = false;
    protected $dateFormat    = 'datetime';
    protected $createdField  = 'created_at';
    protected $updatedField  = 'updated_at';
    protected $deletedField  = 'deleted_at';

    // Validation
    protected $validationRules      = [];
    protected $validationMessages   = [];
    protected $skipValidation       = false;
    protected $cleanValidationRules = true;

    // Callbacks
    protected $allowCallbacks = true;
    protected $beforeInsert   = [];
    protected $afterInsert    = [];
    protected $beforeUpdate   = [];
    protected $afterUpdate    = [];
    protected $beforeFind     = [];
    protected $afterFind      = [];
    protected $beforeDelete   = [];
    protected $afterDelete    = [];

    public function aggregates(): self
    {
        return $this->select('exemplaire.*, document.*, etat.libelle AS etat')
        ->join('etat', 'exemplaire.idEtat = etat.id')
        ->join('document', 'exemplaire.id = document.id')
        ->orderBy('exemplaire.id');
    }
}

File: frontend/MediaTekDocuments/Model/Exemplaire.cs
using System;

namespace MediaTekDocuments.Model
{
    /// <summary>
    /// Classe métier Exemplaire (exemplaire d'une revue)
    /// </summary>
    public class Exemplaire(int numero, DateTime dateAchat, string photo, string idEtat, string idDocument)
	{
		public int Numero { get; set; } = numero;
		public string Photo { get; set; } = photo;
		public DateTime DateAchat { get; set; } = dateAchat;
		public string IdEtat { get; set; } = idEtat;
		public string Id { get; set; } = idDocument;
	}
}
